lib/Test/Pod/CoverageChange.pm

```diff
@@ -0,0 +1,250 @@
+=item * C<$ignored_packages> - arrayref of packages that will be ignored in the checks (optional)
+
+example: ['MyPackage1', 'MyPackage2', 'MyPackage3']
+
+=back
+
+=cut
+
+sub pod_coverage_syntax_ok {
+    my $path                   = shift;
+    my $allowed_naked_packages = shift // {};
+    my $ignored_packages       = shift // [];
+
+    $path             = [$path]             unless ref $path eq 'ARRAY';
+    $ignored_packages = [$ignored_packages] unless ref $ignored_packages eq 'ARRAY';
+
+    check_pod_coverage($path, $allowed_naked_packages, $ignored_packages);
+    check_pod_syntax($path, $ignored_packages);
+}
+
+=head2 check_pod_coverage
+
+Checks POD coverage for all the modules that exist under the given directory.
+Passes the C<$allowed_naked_packages> to L<Test::Pod::CoverageChange/check_allowed_naked_packages>.
+Ignores the packages in the C<$ignored_packages> parameter.
+
+=over 4
+
+=item * C<$path> - path or arrayref of directories to check (recursively)
+
+=item * C<$allowed_naked_packages> - hashref of number of allowed naked subs, keyed by package name (optional)
+
+=item * C<$ignored_packages> - arrayref of packages that will be ignored in the checks (optional)
+
+=back
+
+=cut
+
+sub check_pod_coverage {
+    my $path                   = shift;
+    my $allowed_naked_packages = shift;
+    my $ignored_packages       = shift;
+
+    check_allowed_naked_packages($allowed_naked_packages, $ignored_packages) if keys %$allowed_naked_packages;
+
+    # Check for newly added packages PODs
+    my @ignored_packages = (keys %$allowed_naked_packages, @$ignored_packages);
+    foreach my $package (Test::Pod::Coverage::all_modules(@$path)) {
+        next if any { $_ eq $package } @ignored_packages;
+        pod_coverage_ok($package, {private => []});
+    }
+}
+
+=head2 check_pod_syntax
+
+Check POD syntax for all the modules that exist under the given directory.
+
+=over 4
+
+=item * C<$path> - path or arrayref of directories to check (recursively)
+
+=item * C<$ignored_packages> - arrayref of packages that will be ignored in the checks (optional)
+
+=back
+
+=cut
+
+sub check_pod_syntax {
+    my $path             = shift;
+    my $ignored_packages = shift;
+    my $Test_Builder     = Test::More->builder;
+
+    my @ignored_packages_full_path = ();
+    for (@$ignored_packages) {
+        my $file_path = module_path($_);
+        push @ignored_packages_full_path, $file_path if defined $file_path;
+    }
+
+    my @files_path = File::Find::Rule->file()->name('*.p[m|l]')->in(@$path);
+
+    for my $file_path (@files_path) {
+        next if any { /\Q$file_path/ } @ignored_packages_full_path;
+
+        my $check_result = podchecker($file_path);
+        if ($check_result == POD_SYNTAX_IS_OK) {
+            $Test_Builder->ok(1, sprintf("Pod structure is OK in the file %s.", $file_path));
+        } elsif ($check_result == FILE_HAS_NO_POD) {
+            $Test_Builder->todo_skip(sprintf("There is no POD in the file %s.", $file_path));
```

Comment:
Maybe `->skip` would be nicer?

Reply:
I think we can keep this for one reason, todo_skip will declare the test as failing and TODO, and I think it can be a motivation for our colleagues to add more pods to the subs. what's your opinion about this strategy?